srv/modules/runners/validate.py

```diff
@@ -642,21 +669,27 @@ def pillar(name = None, **kwargs):
     v.pool_creation()
     v.time_server()
     v.fqdn()
-    v.report()
+    v.report(printer)
+    
+    if not has_printer:
+        printer.print_result()
 
     if v.errors:
         return False
 
     return True
 
-def setup():
+def setup(**kwargs):
     """
     Check that initial files prior to any stage are correct
     """
     local = salt.client.LocalClient()
     pillar_data = local.cmd('*' , 'pillar.items', [], expr_form="glob")
+    printer = printer = get_printer(**kwargs)
+
     v = Validate("setup", pillar_data, [])
     v.master_minion()
     v.ceph_version()
-    v.report()
-
+    v.report(printer)
+    
+    printer.print_result()
```

Comment:
What do you think of moving the setting of the printer to the init of the Validate class?  And calling printer.print_result() from v.report()?